src/zcash/ExampleProvingServiceClient.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2017 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "../util.h"
+#include "primitives/transaction.h"
+#include "streams.h"
+#include "utilstrencodings.h"
+#include "version.h"
+#include "zcash/JoinSplit.hpp"
+#include "zcash/Proof.hpp"
+#include "zcash/ProvingServiceClient.hpp"
+
+#include <vector>
+
+#include "libsnark/common/profiling.hpp"
+
+using namespace libzcash;
+
+int main(int argc, char** argv)
+{
+    SetupEnvironment();
+
+    libsnark::inhibit_profiling_info = true;
+    libsnark::inhibit_profiling_counters = true;
+
+    auto p = ZCJoinSplit::Prepared((ZC_GetParamsDir() / "sprout-verifying.key").string(),
+                                   (ZC_GetParamsDir() / "sprout-proving.key").string());
+
+    int32_t numJSDescs = 5;
+    if (argc > 3) {
+        ParseInt32(std::string(argv[3]), &numJSDescs);
+    }
+    if (argc < 3 || numJSDescs <= 0) {
+        std::cerr << "Usage: ExampleProvingServiceClient [provingServiceAddress] [provingServicePubKey] (numJSDescs)" << std::endl;
+        std::cerr << "provingServiceAddress is a ZMQ address, e.g. tcp://localhost:8234" << std::endl;
```

Comment:
given an example of what to write for `provingServicePubKey`?

Reply:
(I myself am not sure how to run the example)

Reply:
Done

Reply:
Don't follow. How to get a public key?